config/config.go

```diff
@@ -138,13 +141,33 @@ func (c *configStruct) StoreEndpointAuth(endpointURL string, email string, token
 	return nil
 }
 
-// SelectEndpoint makes the given endpoint URL the selected one
-func (c *configStruct) SelectEndpoint(endpointURL string) error {
-	ep := normalizeEndpoint(endpointURL)
-	if _, ok := c.Endpoints[ep]; !ok {
+// SelectEndpoint makes the given endpoint the selected one. The argument
+// can either be an alias (that will be used as is) or
+// a URL which will undergo normalization.s
+func (c *configStruct) SelectEndpoint(endpointAliasOrURL string) error {
+
+	if endpointAliasOrURL == "" {
 		return microerror.Mask(endpointNotDefinedError)
 	}
 
+	ep := ""
+
+	// first check if the endpointURL matches an alias.
+	argumentIsAlias := false
+	for key := range c.Endpoints {
+		if endpointAliasOrURL == c.Endpoints[key].Alias {
+			argumentIsAlias = true
+			ep = key
+		}
+	}
```

Comment:
A method like `Endpoints.GetByKey(key string) (string, error)` would make sense to me here. In case there is none, a `notFoundError` should be returned and the according matcher could be used below.